code/game/machinery/kitchen/snackbarmachine.dm

```diff
@@ -1,173 +1,52 @@
 
-/obj/machinery/snackbar_machine/attackby(var/obj/item/weapon/B as obj, var/mob/user as mob)
+	if(istype(B, /obj/item/weapon/storage/pill_bottle))
+		user << "<span class='warning'>This condiment master does not come with a pill dispenser unit built-in.</span>"
+		return 1
 
-	if(istype(B, /obj/item/weapon/reagent_containers/glass))
-		if(src.beaker)
-			user << "A beaker is already loaded into the machine."
-			return
-		src.beaker = B
-		user.drop_item(B, src)
-		user << "You add the beaker to the machine!"
-		src.updateUsrDialog()
-		update_icon()
+	. = ..()
 
-	else
-		..()
+/obj/machinery/chem_master/snackbar_machine/Topic(href, href_list)
 
-/obj/machinery/snackbar_machine/Topic(href, href_list)
-	if(..()) return 1
+	if(..())
+		return 1
 
-	src.add_fingerprint(usr)
 	usr.set_machine(src)
 
 	if(href_list["close"])
 		usr << browse(null, "window=snackbar_machine")
 		usr.unset_machine()
-		return
+		return 1
 
-	if(beaker)
-		var/datum/reagents/R = beaker.reagents
-		if (href_list["analyze"])
-			var/dat = ""
-			dat += "<TITLE>SnackBar Machine</TITLE>Reagent info:<BR><BR>Name:<BR>[href_list["name"]]<BR><BR>Description:<BR>[href_list["desc"]]<BR><BR><BR><A href='?src=\ref[src];main=1'>(Back)</A>"
-			usr << browse(dat, "window=snackbar_machine;size=575x400")
-			return
-
-		else if (href_list["add"])
-
-			if(href_list["amount"])
-				var/id = href_list["add"]
-				var/amount = text2num(href_list["amount"])
-				if (amount < 0) return
-				R.trans_id_to(src, id, amount)
-
-		else if (href_list["addcustom"])
-
-			var/id = href_list["addcustom"]
-			useramount = input("Select the amount to transfer.", 30, useramount) as num
-			useramount = isgoodnumber(useramount)
-			src.Topic(null, list("amount" = "[useramount]", "add" = "[id]"))
-
-		else if (href_list["remove"])
-
-			if(href_list["amount"])
-				var/id = href_list["remove"]
-				var/amount = text2num(href_list["amount"])
-				if (amount < 0) return
-				if(mode)
-					reagents.trans_id_to(beaker, id, amount)
-				else
-					reagents.remove_reagent(id, amount)
-
-		else if (href_list["removecustom"])
-
-			var/id = href_list["removecustom"]
-			useramount = input("Select the amount to transfer.", 30, useramount) as num
-			useramount = isgoodnumber(useramount)
-			src.Topic(null, list("amount" = "[useramount]", "remove" = "[id]"))
-
-		else if (href_list["toggle"])
-			mode = !mode
-
-		else if (href_list["main"])
-			attack_hand(usr)
-			return
-
-		else if (href_list["eject"])
-			if(beaker)
-				beaker:loc = src.loc
-				beaker = null
-				reagents.clear_reagents()
-				update_icon()
-
-		else if (href_list["createbar"])
-			var/obj/item/weapon/reagent_containers/food/snacks/snackbar/SB = new/obj/item/weapon/reagent_containers/food/snacks/snackbar(src.loc)
-			reagents.trans_to(SB, 10)
+	if(href_list["createpill"] || href_list["createpill_multiple"] || href_list["ejectp"] || href_list["change_pill"])
```

Comment:
Same goes for all of this, all of these were made possible when you called the parent with ..() before these return